test/cql-pytest/conftest.py

```diff
@@ -36,34 +34,15 @@ def pytest_addoption(parser):
 # We use scope="session" so that all tests will reuse the same client object.
 @pytest.fixture(scope="session")
 def cql(request):
-    profile = ExecutionProfile(
-        load_balancing_policy=RoundRobinPolicy(),
-        consistency_level=ConsistencyLevel.LOCAL_QUORUM,
-        serial_consistency_level=ConsistencyLevel.LOCAL_SERIAL,
-        # The default timeout (in seconds) for execute() commands is 10, which
-        # should have been more than enough, but in some extreme cases with a
-        # very slow debug build running on a very busy machine and a very slow
-        # request (e.g., a DROP KEYSPACE needing to drop multiple tables)
-        # 10 seconds may not be enough, so let's increase it. See issue #7838.
-        request_timeout = 120)
-    if request.config.getoption('ssl'):
-        # Scylla does not support any earlier TLS protocol. If you try,
-        # you will get mysterious EOF errors (see issue #6971) :-(
-        ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
-    else:
-        ssl_context = None
-    cluster = Cluster(execution_profiles={EXEC_PROFILE_DEFAULT: profile},
-        contact_points=[request.config.getoption('host')],
-        port=int(request.config.getoption('port')),
-        # TODO: make the protocol version an option, to allow testing with
-        # different versions. If we drop this setting completely, it will
-        # mean pick the latest version supported by the client and the server.
-        protocol_version=4,
-        # Use the default superuser credentials, which work for both Scylla and Cassandra
-        auth_provider=PlainTextAuthProvider(username='cassandra', password='cassandra'),
-        ssl_context=ssl_context,
-    )
-    return cluster.connect()
+    # Use the default superuser credentials, which work for both Scylla and Cassandra
+    with cql_session(request.config.getoption('host'),
+        request.config.getoption('port'),
+        request.config.getoption('ssl'),
+        username="cassandra",
+        password="cassandra"
+    ) as session:
+        yield session
+        session.shutdown()
```

Comment:
As I noted elsewhere (maybe you replied elsewhere), I don't understand if session.shutdown() is enough, or you need cluster.shutdown().
If you need cluster.shutdown(), you can make cql_session **itself** a contextmanager instead of a regular function, and the cql fixture can simply do
```
   yield from cql_session(....)
```
(yield from is a new Python feature, the older way to do it would be
```
with cql_session(...) as session:
   yield session
```

Reply:
Oh, but this shutdown() is just for completion. I also made the whole `cql_session` helper a contextmanager, and it properly calls `cluster.shutdown()` after the session is done (e.g. here) - you can see the change in util.py

Reply:
Oh, I missed that... Sorry, Ok. The session.shutdown() is not necessary because cluster.shutdown() does this already, but I guess it doesn't hurt....